### PR TITLE
DRILL-6210: Enhanced test schema utilities

### PR DESCRIPTION
**Pull request by paul-rogers**
The result set loader project enhanced the schema builder used in tests to handle Drill's complex types: maps, unions, lists and repeated lists. The schema builder previously only handled maps.
This PR describes the addition of just this one part of the result set loader.

Also adds a run-time schema improvement: functions to create arrays when testing. Old style:

```
  rowSetBuilder.addRow(1,  new String[] {"foo", "bar"});
```

New style:
```
  rowSetBuilder.addRow(1,  strArray"foo", "bar"));
```

### CODE REVIEW

**Comment by paul-rogers** [comment]
@padma, can you review this?

@ilooner, you might want to take a look also since you've taken an interest in improving our test structures.

**Comment by ppadma** [comment]
This is very useful for testing complex types.  LGTM.  +1.

**Comment by ppadma** [comment]
@Ben-Zvi Boaz, can you do a committer review for this PR and include it in this week's batch commit ?

File: exec/java-exec/src/test/java/org/apache/drill/test/rowSet/schema/SchemaBuilder.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+ *          .addType(MinorType.INT)
+ *          .addMap()
+ *            .add("h1", MinorType.INT)
+ *            .buildNested()
+ *          .addList()
+ *            .addType(MinorType.BIGINT)
+ *            .buildNested()
+ *          .build()
+ *        .addArray("i", MinorType.BIGINT)
+ *        .addRepeatedList("k")
```
Comment (Ben-Zvi) [inline exec/java-exec/src/test/java/org/apache/drill/test/rowSet/schema/SchemaBuilder.java]: "k" should be "j" ....

**Comment by paul-rogers** [comment]
Fixed the typo.

@Ben-Zvi, thanks for pointing out the unit tests: I do have test file in my branch; forgot to include it with the PR. Just did.

**Comment by paul-rogers** [comment]
@Ben-Zvi, the changes look right. Old style was to have a "buildFoo" method for the nested structure "foo". The new style is to have a "resumeBar" method for the outer structure "bar".

If you have trouble, an alternative is to commit @ppadma's PR. Then I can immediately rebase and make the required fixes to her tests.

**Comment by paul-rogers** [comment]
Rebased on latest master.

**Pull request closed without merging**